api/types/plugin_data.go

```diff
@@ -126,17 +119,29 @@ func (r *PluginDataV3) String() string {
 	return fmt.Sprintf("PluginData(kind=%s,resource=%s,entries=%d)", r.GetSubKind(), r.GetName(), len(r.Spec.Entries))
 }
 
+// setStaticFields sets static fields
+func (o *PluginDataV3) setStaticFields() {
+	o.Kind = KindPluginData
+	o.Version = V3
+}
+
 // CheckAndSetDefaults checks and sets default values for PluginData.
 func (r *PluginDataV3) CheckAndSetDefaults() error {
+	r.setStaticFields()
+	if r.Spec.Entries == nil {
+		r.Spec = PluginDataSpecV3{
+			Entries: make(map[string]*PluginDataEntry),
+		}
+	}
```

Comment:
Just because the `Entries` field is nil, we overwrite the whole `Spec`? Even if `Entries` is currently the only field, the checks already in place should remain correct even after a new field happens to be added.